presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py

```diff
@@ -46,24 +50,182 @@ def __init__(
+        """
+
+        nlp_recognizers = [
+            rec
+            for rec in recognizers
+            if isinstance(rec, SpacyRecognizer) and rec.supported_language == language
+        ]
+
+        # Case 1: NLP recognizer is not in the list of recognizers
+        if not nlp_recognizers:
+            warning_text = (
+                f"NLP recognizer (e.g. SpacyRecognizer, StanzaRecognizer) "
+                f"is not in the list of recognizers "
+                f"for language {language}. "
+                f"Adding the default {SpacyRecognizer.__name__} to the list."
+                f"If you wish to remove the NLP recognizer, "
+                f"define it as `enabled=false`."
+            )
+            logger.warning(warning_text)
+            warnings.warn(warning_text)
+            if nlp_engine:
+                recognizers.append(
+                    SpacyRecognizer(
+                        supported_language=language,
+                        supported_entities=nlp_engine.get_supported_entities(),
+                    )
+                )
+            else:
+                recognizers.append(SpacyRecognizer(supported_language=language))
+            return
+
+        # Case 2: There are multiple NLP recognizers for this language, throw error
+        if len(nlp_recognizers) > 1:
+            raise ValueError(
+                f"Multiple NLP recognizers for language {language} "
+                f"found in the configuration. "
+                f"Please remove the duplicates."
+            )
+
+        # Case 3: There is a mismatch between the NLP Engine and the NLP Recognizer
+        nlp_recognizer = nlp_recognizers[0]
+        expected_nlp_recognizer_cls = RecognizerRegistry.get_nlp_recognizer(nlp_engine)
+        if nlp_recognizer.__class__ != expected_nlp_recognizer_cls:
+            raise ValueError(
+                f"There is a mismatch between the NLP Engine defined "
+                f"({nlp_engine.__class__.__name__}),"
+                f"and the configured NLP recognizer "
+                f"({nlp_recognizer.__class__.__name__})."
+                f"Make sure the NLP recognizer is aligned with the "
+                f"NLP engine and that all others are removed/disabled."
+            )
+
+    @staticmethod
+    def __remove_disabled_nlp_recognizers(
+        recognizers: List[EntityRecognizer],
+        recognizers_conf: Dict[str, Any],
+        language: str,
+    ):
+        """
+        Remove recognizers that are disabled in the configuration.
+
+        Goes through the recognizer conf provided by the user,
+        and checks if a recognizer for a given language is disabled.
+        If yes, it removes it from the recognizers list
+        (as some are not removed in the previous step).
+        """
+
+        disabled = [
+            rec_conf
+            for rec_conf in recognizers_conf
+            if RecognizerListLoader.is_recognizer_enabled(rec_conf) is False
```

Comment:
if not RecognizerListLoader.is_recognizer_enabled(rec_conf)

Reply:
fixed